### PR TITLE
feat: Drop `adminClient` (2/2)

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Follow on from https://github.com/theopensystemslab/planx-new/pull/2356 - please see this PR for context
 - Removes `adminClient` from API codebase
 - Does not remove `HASURA_GRAPHQL_ADMIN_SECRET` from the API codebase, as this is still required for the Hasura Metadata and Schema APIs

### CODE REVIEW

File: api.planx.uk/editor/findReplace.ts [hunk 1]
```diff
@@ -136,14 +146,18 @@ const findAndReplaceInFlow = async (
       }
 
       // if matches, proceed with mutation to update flow data
-      const response = await adminClient.request(
+      const { client: $client } = getClient();
```
Comment (DafyddLlyr) [inline api.planx.uk/editor/findReplace.ts:149]: Not massively fond of this syntax but it's somewhat of a solution to this issue - https://github.com/theopensystemslab/planx-new/pull/2297#discussion_r1357004769

File: api.planx.uk/helpers.ts [hunk 2]
```diff
@@ -83,53 +87,42 @@ const createAssociatedOperation = async (flowId: Flow["id"]) => {
...
   );
 
-  return data?.insert_operations_one;
+  return data?.operation;
 };
 
+interface PublishedFlows {
+  flow: {
+    publishedFlows: {
+      // TODO: use FlowGraph from planx-core here
```
Comment (DafyddLlyr) [inline api.planx.uk/helpers.ts:96]: This `TODO` is intentionally left here - making this change has pretty messy cascading effects that would add a lot of noise to this PR.

I'll make a GitHub issue describing the required change here.

File: api.planx.uk/helpers.ts [hunk 3]
```diff
@@ -83,53 +87,42 @@ const createAssociatedOperation = async (flowId: Flow["id"]) => {
...
-    {
-      id,
-      created_at,
-    },
+    { id },
   );
 
-  return data.flows_by_pk.published_flows?.[0]?.data;
+  const mostRecent = flow?.publishedFlows?.[0]?.data;
+  if (!mostRecent) throw Error(`Published flow not found for flow ${id}`);
```
Comment (DafyddLlyr) [inline api.planx.uk/helpers.ts:123]: Generally we do error catching outside these little request functions but making it once here actually is much cleaner.

Going forward to would be nice to handle this in a much more standardised method - I actually think Apollo has some really nice patterns for this which may be something to consider.

File: api.planx.uk/editor/copyFlow.test.ts [hunk 4]
```diff
@@ -20,7 +20,7 @@ beforeEach(() => {
     name: "InsertFlow",
     matchOnVariables: false,
     data: {
-      insert_flows_one: {
+      newFlow: {
```
Comment (jessicamcinchak) [inline api.planx.uk/editor/copyFlow.test.ts:23]: nit: for other `insert_` mutations, it seems you are't prepending `new` - can this just be `flow` for consistency?

**Comment by DafyddLlyr** [comment]
Thanks for catching these naming issues!

File: api.planx.uk/helpers.ts (hunk 2 above)
Comment (DafyddLlyr) [reply api.planx.uk/helpers.ts:96]: Issue logged here - https://github.com/theopensystemslab/planx-new/issues/2405

**Pull request merged**